Source/WebKit/Shared/AuxiliaryProcess.cpp

```diff
@@ -250,4 +250,33 @@ void AuxiliaryProcess::didReceiveMemoryPressureEvent(bool isCritical)
 
 #endif // !PLATFORM(COCOA)
 
+bool AuxiliaryProcess::allowsFirstPartyForCookies(const URL& firstParty)
+{
+    // FIXME: This should probably not be necessary. If about:blank is the first party for cookies,
+    // we should set it to be the inherited origin then remove this exception.
+    if (firstParty.isAboutBlank())
+        return true;
+
+    if (firstParty.isNull())
+        return true; // FIXME: This shouldn't be allowed.
+
+    return false;
```

Comment:
This isn't really "false" - this value should indicate that another check is needed.  I think it would be better if this function required a HashSet and passed it to the other AuxiliaryProcess::allowsFirstPartyForCookies instead of returning a misleading value here.